HeterogeneousCore/CUDAUtilities/interface/HistoContainer.h

```diff
@@ -86,15 +82,20 @@ namespace cms {
       uint32_t *off = (uint32_t *)((char *)(h) + offsetof(Histo, off));
       size_t wss = Histo::wsSize();
       assert(wss > 0);
-      CubDebugExit(cub::DeviceScan::InclusiveSum(ws, wss, off, off, Histo::totbins(), stream));
+      auto nthreads = 512+256;
+      auto nblocks = (Histo::totbins() + nthreads - 1) / nthreads;
+      assert(nblocks<=1024);
+      cudaCheck(cudaMemsetAsync(ws, 0, 4, stream));
```

Comment:
In principle one could find a way to save this (zeroing in the previous kernel, or similar)

Reply:
The other possibility would be to use cooperative groups with [inter-grid synchronization](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#grid-synchronization-cg).
Do you think it is worth considering ?

Reply:
1) I my opinion not for the current size of the histogram
2) I should check,  cooperative groups had some similarity with dynamic parallelism
I will anyhow give a try and check performance (btw nvprof is crashing in 10.2)

Reply:
I suspect that to use inter-grid synchronization one must ensure that all blocks are scheduled at the same time. This is not obvious in particular if the accelerator is shared among different processes/threads.
most probably cudaLaunchCooperativeKernel ensures that.
I will try to experiment.

Reply:
just launching as is 
/home/innocent/PSreloaded/CMSSW_11_1_Patatrack_X_2020-05-13-2300/src/HeterogeneousCore/CUDAUtilities/interface/launch.h, line 141:
cudaCheck(cudaLaunchCooperativeKernel( (const void*)kernel, config.gridDim, config.blockDim, (void**)pointers, config.sharedMem, config.stream));
cudaErrorCooperativeLaunchTooLarge: too many blocks in cooperative launch

Reply:
ok, so the number or blocks that can cooperate is too small for our use case